From c4c4fd2a816937f938b1ab370c2bde54f342de0a Mon Sep 17 00:00:00 2001
From: Gareth Davidson <gaz@bitplane.net>
Date: Sun, 6 Sep 2026 08:48:46 +0100
Subject: [PATCH 1/6] Fix array dimension list traversal

Start the dimension walk at the supplied CArrayDim. Taking the address of
the pointer parameter treats a stack slot as a dimension record and writes
outside it while updating total_cnt.
---
 Src/PrsVar.HC | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/Src/PrsVar.HC b/Src/PrsVar.HC
index 5d74704..7893867 100644
--- a/Src/PrsVar.HC
+++ b/Src/PrsVar.HC
@@ -277,14 +277,8 @@ U0 PrsArrayDims(CCmpCtrl *cc,I64 mode,CArrayDim *dim)
       }
       tmpad=MAlloc(sizeof(CArrayDim));
       tmpad->next=NULL;
-      tmpad1=&dim;
+      tmpad1=dim;
       do {
-      //
-      //Nroot Fix,getting addres of &dim is an 8byte ptr,SO DONT WRITE INTO IT IT
-      // [RBP]
-      // [RBP-8] &dim(writing a member may corrupt RBP)
-      //
-      if(&dim!=tmpad1)
         tmpad1->total_cnt*=j;
 	if (!tmpad1->next) {
 	  tmpad1->next=tmpad;
-- 
2.53.0
